src/gt4sd/algorithms/conditional_generation/paccmann_rl/implementation.py

```diff
@@ -92,17 +92,18 @@ def get_smiles_from_latent(self, latent: torch.Tensor) -> List[str]:
             search=SamplingSearch(temperature=self.temperature),
         )
 
-        # decode SELFIES
-        selfies = [
+        molecules = [
             self.smiles_language.token_indexes_to_smiles(generated_molecule.tolist())
             for generated_molecule in iter(generated_molecules)
         ]
 
         # convert SELFIES to SMILES
-        smiles = [
-            self.smiles_language.selfies_to_smiles(a_selfies) for a_selfies in selfies
-        ]
-        return smiles
+        if "selfies" in self.smiles_language.name:
```

Comment:
can't think of a better way so we will live with this

Reply:
indeed, if we move to the latest `pytoda` this will be trivial and safe, but for the old version, it is what it is :/